src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,146 @@
+    private Entry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Entry[DEFAULT_CAPACITY];
+    }
+
+    static class Entry<K, V> {
+        private final K key;
+        private V value;
+        private Entry<K, V> next;
+
+        public Entry(K key, V value, Entry<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public final K getKey() {
+            return key;
+        }
+
+        public final V getValue() {
+            return value;
+        }
+
+        public final boolean equals(Object o) {
+            if (!(o instanceof Map.Entry)) {
+                return false;
+            }
+            Map.Entry<K, V> entry = (Map.Entry<K, V>) o;
+            Object key1 = getKey();
+            Object key2 = entry.getKey();
+            if (Objects.equals(key1, key2)) {
+                Object value1 = getValue();
+                Object value2 = entry.getValue();
+                return Objects.equals(value1, value2);
+            }
+            return false;
+        }
+
+        public final int hashCode() {
+            return (key == null ? 0 : Math.abs(key.hashCode()));
+        }
+    }
+
+    private int indexFor(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % DEFAULT_CAPACITY);
+    }
+
+    private void putForNullKey(V value) {
+        for (Entry<K, V> entry = table[0]; entry != null; entry = entry.next) {
+            if (entry.key == null) {
+                entry.value = value;
+                return;
+            }
+        }
+        addEntry(null, value, 0);
+    }
+
+    private void addEntry(K key, V value, int index) {
+        Entry<K, V> entry = table[index];
+        table[index] = new Entry<>(key, value, entry);
+        if (size++ >= threshold) {
+            resize(2 * table.length);
+        }
+    }
+
+    private void resize(int newCapacity) {
+        Entry<K, V>[] newTable = new Entry[newCapacity];
+        transfer(newTable);
+        table = newTable;
+        threshold = (int) (newCapacity * LOAD_FACTOR);
+    }
+
+    private void transfer(Entry<K, V>[] newTable) {
+        Entry<K, V>[] src = table;
+        for (int j = 0; j < src.length; j++) {
+            Entry<K, V> entry = src[j];
+            if (entry != null) {
+                src[j] = null;
+            }
+            while (entry != null) {
+                Entry<K, V> next = entry.next;
+                int i = indexFor(entry.key);
+                entry.next = newTable[i];
+                newTable[i] = entry;
+                entry = next;
+            }
+        }
+    }
+
+    private V getForNullKey() {
+        for (Entry<K, V> entry = table[0]; entry != null; entry = entry.next) {
+            if (entry.key == null) {
+                return entry.value;
+            }
+        }
+        return null;
+    }
 
     @Override
     public void put(K key, V value) {
```

Comment:
logic in this method should be like this:
1. Check if we should resize table, if true make resize
2. Find an index of a bucket based on the key's hash
3. if a bucket is empty create a new Entry and fill this bucket with its first value
4. if a bucket is not empty go through all chain of entries in this bucket: 
    - if keys equal the change value of an entry
    - if not go to the next entry in a chain till we can, if the next element is null add a new Entry in the end of a chain